### PR TITLE
MESMER-X: more restructuring, define module-level variables

### PR DESCRIPTION
**Pull request by mathause**
I extracted some changes from #466
- more reformatting
- created f-strings
- import `scipy` instead `scipy.stats`
   ```diff
   -import scipy.stats as ss
   +import scipy as sp
   ```
- define module-level variables, which allow to write some code more concisely
  ```
  _DISCRETE_DISTRIBUTIONS = ...
  _CONTINUOUS_DISTRIBUTIONS = ...
  _ALL_DISTRIBUTIONS = ...
  _NUMBERS = ...
  ```

---

Just for reference - here is how I did this:

<details>

```bash

# squash the mesmer_x_first_code_updates branch (to avoid cherry-pick conflict)
# https://stackoverflow.com/a/25357146

git switch mesmer_x_first_code_updates
git reset --soft $(git merge-base main HEAD)
git commit -m "MESMER-X: first code updates"
git push origin HEAD --force


# cherry pick onto new branch
git co -b mx_more_trivial_updates
git cherry-pick mesmer_x_first_code_updates

# unstage the cherry-picked commit
git reset --soft 02169332a342275bf7e12db0dade60b5312ecc93
git restore --staged .

# interactive (patch) add the relevant parts of the code
git add -u -p
```

</details>

### CODE REVIEW

**Review by 925209391 — APPROVED** [review]

**Comment by mathause** [comment]
@yquilcaille & @veni-vidi-vici-dormivi FYI & if you are interested but I am also happy to merge these changes without review

File: mesmer/mesmer_x/train_utils_mesmerx.py [hunk 1]
```diff
@@ -10,11 +10,16 @@
...
-# import sympy as sp
+import scipy as sp
 import xarray as xr
 
+_DISCRETE_DISTRIBUTIONS = sp.stats._discrete_distns._distn_names
+_CONTINUOUS_DISTRIBUTIONS = sp.stats._continuous_distns._distn_names
+
+_ALL_DISTRIBUTIONS = _DISCRETE_DISTRIBUTIONS + _CONTINUOUS_DISTRIBUTIONS
+
+_NUMBERS = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"]
```
Comment (veni-vidi-vici-dormivi) [inline mesmer/mesmer_x/train_utils_mesmerx.py:21]: Maybe rather _DIGITS?

**Review by veni-vidi-vici-dormivi — APPROVED** [review]

File: mesmer/mesmer_x/train_utils_mesmerx.py (hunk 1 above)
Comment (mathause) [reply mesmer/mesmer_x/train_utils_mesmerx.py:21]: Yes, that's a better name! Thanks

**Review by 925209391 — APPROVED** [review]

File: mesmer/mesmer_x/train_l_distrib_mesmerx.py [hunk 2]
```diff
@@ -141,18 +141,13 @@ def xr_train_distrib(
...
-            + target_name
-            + " with the expression "
-            + expr
-            + ": "
-            + str(np.round(100.0 * (igp + 1) / gridpoints.size, 1))
-            + "%",
-            end="\r",
-        )
+        fraction = (igp + 1) / gridpoints.size
+        print(f"{fraction:0.1%}", end="\r")
```
Comment (yquilcaille) [inline mesmer/mesmer_x/train_l_distrib_mesmerx.py:150]: In my experience, it is useful to print what is being trained. For instance, it is quite often that i was training different expressions on different kernels, so it is nice to remember where is trained what. Though, i agree that it is longer.

File: mesmer/mesmer_x/train_l_distrib_mesmerx.py [hunk 3]
```diff
@@ -271,9 +262,9 @@ def __init__(
         self,
         data_targ,
         data_pred,
-        expr_fit,  # TODO: replace by Expression instance
-        data_targ_addtest=None,
-        data_preds_addtest=None,
+        expr_fit,
+        data_targ_addtest=None,  # TODO: rename to data_targ_validation?
```
Comment (yquilcaille) [inline mesmer/mesmer_x/train_l_distrib_mesmerx.py:266]: This extra data is not exactly about validation, but really about checking whether the trained conditional distribution works with this data as well. Maybe verification?

**Review by yquilcaille — COMMENTED** [review]
I agree with the changes. I commented some minor aspects, but I think that if you deem them as useful, they could be added in an ensuing commit.

File: mesmer/mesmer_x/train_l_distrib_mesmerx.py (hunk 2 above)
Comment (mathause) [reply mesmer/mesmer_x/train_l_distrib_mesmerx.py:150]: It's printed above the fraction but yes could make it a one-liner again

File: mesmer/mesmer_x/train_l_distrib_mesmerx.py (hunk 3 above)
Comment (mathause) [reply mesmer/mesmer_x/train_l_distrib_mesmerx.py:266]: Thanks! Now updated in #466

**Pull request merged**